### DISCUSSION TITLE
Possible way of trigging a method anytime a screen changes?

### DISCUSSION THREAD

**Discussion by EladKarni** [wix/react-native-navigation#7263] [Q&A] [answered]
So I have an app with several bottom navigation tabs. If the user is navigating within one of these tabs (selects that tab, and navigates around within it), I'd like to call a function each time they move screens inside there. Is that possible?

My current way of solving this would probably be to run something on each update, and check that the tab index is still pointing to the one I want to keep track of.

Any ideas would be greatly appreciated

**Comment by hugo-chq** [comment, marked as answer]
`registerComponentDidAppearListener` might be a good starting point
```
Navigation.events().registerComponentDidAppearListener(({componentId}) => {
  console.log('componentId', componentId);
});
```
it should log the screen name when it changes

**Reply by swabbass** [reply]
Good answer, thx.
@EladKarni you might need to determine which tab is currently selected, it is fairly easy, have a look at the docs and report back if you still need help.
